File: src/0027-Remove-Element/remove_element.py
from typing import List

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        k: int = 0
        nums_len: int = len(nums) - 1
        while nums_len >= 0:
            if nums[nums_len] == val:
                nums.pop(nums_len)
            else:
                k +=1
            nums_len -= 1
        return k
        

if __name__ == "__main__":
    nums: List[int] = [3,2,2,3]
    val: int = 3
    solution = Solution()
    
    print(solution.removeElement(nums,val))

File: src/0027-Remove-Element/remove_element.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int removeElement(std::vector<int>& nums, int val) {
        int k = 0;
        int nums_len = nums.size() - 1;
        while (nums_len >= 0){
            if (nums[nums_len] == val) {
                nums.erase(nums.begin() + nums_len);
            }
            else {
                k++;
            }
            nums_len--;
        }
        return k;
    }
};

int main() {
    std::vector<int> nums = {3,2,2,3};
    int val = 3;
    
    Solution solution;

    std::cout << solution.removeElement(nums, val) << " ";
    for (int i : nums) {
        std::cout << i << " ";
    }

    return 0;
}